src/Drupal/DrupalExtension/Context/DrupalContext.php

```diff
@@ -183,32 +183,135 @@ public function getTableRow(Element $element, $search)
         throw new \Exception(sprintf('Failed to find a row containing "%s" on the page %s', $search, $this->getSession()->getCurrentUrl()));
     }
 
+  /**
+   * Retrieve table rows containing specified text from a given element.
+   *
+   * @param Element $element
+   *   \Behat\Mink\Element\Element object.
+   * @param string $search
+   *   The text to search for in the table row.
+   *
+   * @return array of \Behat\Mink\Element\NodeElement
+   *
+   * @throws \RuntimeException
+   */
+  public function getTableRows(Element $element, $search): array {
+    $rows = $element->findAll('css', 'tr');
+    if (empty($rows)) {
+      throw new \RuntimeException(sprintf('No rows found on the page %s', $this->getSession()->getCurrentUrl()));
+    }
+    $found_rows = [];
+    foreach ($rows ?: [] as $row) {
+      if (strpos($row->getText(), $search) !== FALSE) {
+        array_push($found_rows, $row);
+      }
+    }
```

Comment:
@kumikoono Let's use a guard clause:

```suggestion
    foreach ($rows ?: [] as $row) {
      if (strpos($row->getText(), $search) === FALSE) {
        continue;
      }
      array_push($found_rows, $row);
    }
```